libraries/AP_Mount/AP_Mount_Backend.cpp

```diff
@@ -799,6 +799,8 @@ uint16_t AP_Mount_Backend::get_gimbal_device_flags() const
 {
     const uint16_t flags = (get_mode() == MAV_MOUNT_MODE_RETRACT ? GIMBAL_DEVICE_FLAGS_RETRACT : 0) |
                            (get_mode() == MAV_MOUNT_MODE_NEUTRAL ? GIMBAL_DEVICE_FLAGS_NEUTRAL : 0) |
+                           (_yaw_lock ? GIMBAL_DEVICE_FLAGS_YAW_LOCK : 0) |
```

Comment:
I think this is mixing up the **target** angles or rates vs the **actual** angles or rates.  "get_gimbal_device_flags()" is only used to fill in the GIMBAL_DEVICE_ATTITUDE_STATUS message's flags field that is sent to the GCS.  Even if the target are in earth-frame, the angles and rates we're reporting are always in body-frame.  I could be wrong but I think this change would lead to us sometimes tell the GCS the angles and rates are earth-frame when they're not.